iron-form.html

```diff
@@ -130,6 +173,15 @@
         valid = el.validate() && valid;
       }
       return valid;
+    },
+
+    _useValue: function(el) {
+      // Checkboxes and radio buttons should only use their value if they're checked.
+      if (el.type !== 'checkbox' && el.type !== 'radio') {
```

Comment:
This almost seems like something that could be implemented by the input. For example, the input behavior could have a `hasUsableValue` method that can always return true by default, but can be overridden by `checkbox` and `radio` types to use a different heuristic (e.g., `return this.checked`).